CONTRIBUTING.md

```diff
@@ -601,3 +601,41 @@ We communicate with various integrators if they'd like release-blocking QA testi
 [3]:https://github.com/tendermint/tendermint/blob/main/types/results.go#L47-L54
 [4]:https://github.com/osmosis-labs/cosmos-sdk/blob/5c9a51c277d067e0ec5cf48df30a85fae95bcd14/store/rootmulti/store.go#L430
 [5]:https://github.com/osmosis-labs/cosmos-sdk/blob/5c9a51c277d067e0ec5cf48df30a85fae95bcd14/store/types/commit_info.go#L40
+
+## Common Security Considerations
+
+There are several security patterns that come up frequently enough to be synthesized into general rules of thumb. While the high level risks are appchain agnostic, the details are mostly tailored to contributing to Osmosis. This is, of course, not even close to a complete list – just a few considerations to keep in mind.
+
+### Rounding Behavior
+
+As a general rule of thumb, for DEX-related operations, we should be rounding in whichever direction is in the pool’s favor. This is to ensure that no single operation can output more value than was passed in and thus allow for a pool to be drained.
+
+Note that in many cases, such attacks are made unprofitable by fees, but to ensure that these attack vectors are never exposed in the first place, we need to ensure we round properly at each step.
+
+#### Examples
+
+- Round input tokens required for a swap up
+
+- Round output tokens from a swap down
+
+- Round charged fees for swaps up
+
```

Comment:
I think we should mention that this is CL specific, even better when we can have code line!

For example, https://github.com/osmosis-labs/osmosis/blob/fcce61864f/x/gamm/pool-models/balancer/pool.go#L573 here for a Balancer pool, we are rounding down the input tokens

Reply:
This is not just for CL pools! Scared me there for a sec with that example :) The code you linked rounds up (notice that it does `.Ceil()` before `.TruncateInt`)